cmd/controller/collectdiskinfo_test.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2018 OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"io/ioutil"
+	"os/exec"
+	"strconv"
+	"strings"
+	"testing"
+
+	apis "github.com/openebs/node-disk-manager/pkg/apis/ndm/v1alpha1"
+	"github.com/openebs/node-disk-manager/pkg/udev"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCollectDiskInfo(t *testing.T) {
```

Comment:
Can we have unit tests written table driven format?

Reply:
we can have but this test case is system dependent. If only os disk is there then You will get only one details and it will happen for most of the cases so I put just one test case.

Reply:
@shovanmaity let us make use of TDD & then make use of sub tests for UT.
We need to avoid all instances of being dependant on any binaries during UT.
We need to make UT independent of system.
UT should only be concerned of testing the code you have written i.e. the business logic.
UT should target one function at a time.
UT can mock the target function's dependants by making use of go interfaces.

Reply:
changed ut as sub tests.